### PR TITLE
Promote to Channel pipeline

### PR DESCRIPTION
**Pull request by ukff**

### CODE REVIEW

File: docs/workflows.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+### Overview
+
+## Promote to channel
+
+Goal of workflow is to register module by using btp manger template file. Workflow takes tag and downloads module template file from realase on github, and place it to according folder in module folder inside kyma repository, by creating pull request. The pull request need to by approved manually.
+
+Inputs:
+- Release tag: if not specified workflow will take tag from latest avaiablie realase on github and use it. Tag can also be specified directly, the workflow will validate if given tag exists on any relase and if yes, then use it, otherwise the workflow will break execution.
+- Channel: there are three options as value: alpha, fast, regular. This options corresponds to folders in modules folder at Kyma repo.
```
Comment (IwonaLanger) [inline docs/workflows.md:9]:
```suggestion
- Channel - There are three release channels to choose from: alpha, fast, and regular. The options correspond to the subfolders in the module folder at the Kyma repository. 
```

File: docs/workflows.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+### Overview
+
+## Promote to channel
+
+Goal of workflow is to register module by using btp manger template file. Workflow takes tag and downloads module template file from realase on github, and place it to according folder in module folder inside kyma repository, by creating pull request. The pull request need to by approved manually.
+
+Inputs:
```
Comment (IwonaLanger) [inline docs/workflows.md:7]:
```suggestion
If you want to control the workflow, you can set the following inputs in the GitHub UI:
```

File: .github/workflows/promote.yml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                  -H "X-GitHub-Api-Version: 2022-11-28" \
+                  https://api.github.com/repos/$ORG/$BTP_MANAGER_REPO/tags | jq -r '.[] | .name')
+          
+          if  echo $tags | tr " " '\n' | grep -F -q -x ${{ inputs.releaseTag }}; then
+              echo "TAG=${{ inputs.releaseTag }}" >> $GITHUB_ENV
+              echo "found tag"
+              exit 0
+          fi
+
+          echo "no found: ${{ inputs.releaseTag }}"
```
Comment (jaroslaw-pieszka) [inline .github/workflows/promote.yml:60]:
```suggestion
          echo "not found: ${{ inputs.releaseTag }}"
```

File: .github/workflows/promote.yml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      shell: bash
+      run: |
+          tags=$(curl \
+                  -H "Accept: application/vnd.github+json" \
+                  -H "X-GitHub-Api-Version: 2022-11-28" \
+                  https://api.github.com/repos/$ORG/$BTP_MANAGER_REPO/tags | jq -r '.[] | .name')
+          
+          if  echo $tags | tr " " '\n' | grep -F -q -x ${{ inputs.releaseTag }}; then
+              echo "TAG=${{ inputs.releaseTag }}" >> $GITHUB_ENV
+              echo "found tag"
```
Comment (jaroslaw-pieszka) [inline .github/workflows/promote.yml:56]:
```suggestion
              echo "tag found"
```

File: .github/workflows/promote.yml [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+      working-directory: kyma
+      run: |
+        git remote set-url origin https://${{ secrets.ACCESS_TOKEN }}@github.com/$ORG/$KYMA_REPO.git
+        git push --set-upstream origin $BRANCH_NAME -f
+
+    - name: Create PR if needed
+      working-directory: kyma
+      shell: bash
+      run: |
+        prs=$(gh pr list -R https://github.com/$ORG/$KYMA_REPO/ -A ukff --state open --json headRefName | jq -r '.[] | .headRefName')
```
Comment (jaroslaw-pieszka) [inline .github/workflows/promote.yml:107]: Your id is hard-wired here, I am afraid it should not.

File: .github/workflows/promote.yml [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+       
+        if  echo $prs | tr " " '\n' | grep -F -q -x $BRANCH_NAME; then
+            echo "open pr already exists, no need to create new one, pr will be updated by push from previous step"
```
Comment (jaroslaw-pieszka) [inline .github/workflows/promote.yml:110]:
```suggestion
            echo "opened PR already exists, no need to create a new one, PR will be updated by push from previous step"
```

File: .github/workflows/promote.yml [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    - name: Validate Given Relase Tag
+      if: inputs.releaseTag != 'use latest available release'
+      shell: bash
+      run: |
+          tags=$(curl \
+                  -H "Accept: application/vnd.github+json" \
+                  -H "X-GitHub-Api-Version: 2022-11-28" \
+                  https://api.github.com/repos/$ORG/$BTP_MANAGER_REPO/tags | jq -r '.[] | .name')
+          
+          if  echo $tags | tr " " '\n' | grep -F -q -x ${{ inputs.releaseTag }}; then
```
Comment (jaroslaw-pieszka) [inline .github/workflows/promote.yml:54]: Why do we replace spaces with new lines? Aren't the tags returned already line separated?

File: .github/workflows/promote.yml [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+        - regular
+        required: true
+
+jobs:
+  promote:
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: Use default Release Tag
```
Comment (jaroslaw-pieszka) [inline .github/workflows/promote.yml:34]: Possible we can literally mention in the step name, that `default` is `latest`.
E.g. : -name: Use default (latest) Release Tag

File: .github/workflows/promote.yml [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+          exit 1
+
+    - name: Set branch name
+      run: echo "BRANCH_NAME=$BTP_MANAGER_REPO-$TAG" >> $GITHUB_ENV
+
+    - name: Download module template 
+      run: |
+          gh api \
+            -H "Accept: application/vnd.github+json" \
+            https://github.com/$ORG/$BTP_MANAGER_REPO/releases/download/$TAG/$TEMPLATE_NAME >> $TEMPLATE_NAME
```
Comment (jaroslaw-pieszka) [inline .github/workflows/promote.yml:70]: Is `>>` safe here? Why not `>` only.

File: .github/workflows/promote.yml (hunk 7 above)
Comment (jaroslaw-pieszka) [reply .github/workflows/promote.yml:54]:
```
➜  github-actions tags=$(curl \ 
                   -H "Accept: application/vnd.github+json" \
                   -H "X-GitHub-Api-Version: 2022-11-28" \
                   https://api.github.com/repos/kyma-project/btp-manager/tags | jq -r '.[] | .name')
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   965  100   965    0     0   3287      0 --:--:-- --:--:-- --:--:--  3362
➜  github-actions echo $tags
0.1.0
0.0.1
```

Comment (jaroslaw-pieszka) [reply .github/workflows/promote.yml:55]: But nevertheless this replacement does not hurt.

**Review by jaroslaw-pieszka — APPROVED** [review]

**Review by jaroslaw-pieszka — DISMISSED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**